### ISSUE TITLE
Error! - Incompatible operand typres

### ISSUE THREAD

**Issue by ddaddy** [core-plot/core-plot#446]
I've successfully used this through `Carthage` before, but today I tried to rebuild it using:
```
carthage update --use-xcframeworks --platform ios core-plot
```
and it fails to build with the following error:

> core-plot/framework/Source/CPTAnimation.m:339:91: error: incompatible operand types ('Class' and 'NSNull * _Nonnull') [-Werror]
> CPTAnimationValueClassKey: valueClass ? valueClass : [NSNull null],

If I open up the checked out project and try to build it in Xcode, I get the same error in `CPTAnimation.m` on line 339

<img width="1418" alt="Screenshot 2021-05-26 at 17 14 59" src="https://user-images.githubusercontent.com/1425808/119695288-ee99a680-be45-11eb-9de1-98f6edc0aa19.png">

I'm not sure if this is something introduced with Xcode 12.5

**Comment by marcusthornton** [comment]
Hi, I can confirm I am having the exact same problem, and have made the suggest edits as a quick work around

**Comment by mrcmd** [comment]
This is fixed in the release-2.4 branch: https://github.com/core-plot/core-plot/commit/b4cf7a7b468c9601e2d4ae9f3a355ee26269978d
